test/framework/ownerreference_helpers.go

```diff
@@ -382,6 +382,22 @@ func setClusterPause(ctx context.Context, cli client.Client, clusterKey types.Na
 	Expect(cli.Patch(ctx, cluster, pausePatch)).To(Succeed())
 }
 
+// SkipClusterObjectsWithoutNameContains is used in e2e tests where the owner graph
+// gets queried to filter out cluster-wide objects which don't have the clusterName
+// in their object name. This avoids assertions on objects which are part of in-parallel
+// running tests like ExtensionConfig.
```

Comment:
Can we change clusterName to `s` or `str` (+update the godoc accordingly

(we can still mention it as an example)